config/config-api/src/com/thoughtworks/go/domain/PipelineGroups.java

```diff
@@ -158,18 +158,17 @@ public void addError(String fieldName, String message) {
         configErrors.add(fieldName, message);
     }
 
-    public HashSet<MaterialConfig> getAllUniquePostCommitSchedulableMaterials() {
-        HashSet<MaterialConfig> materialConfigs = new HashSet<MaterialConfig>();
-        Set<Map> uniqueMaterials = new HashSet<Map>();
```

Comment:
@srinivasupadhya It was a `Set<Map>` because it was storing & comparing `sqlCriteria` which is a `Map`. I cannot recall the exact reason for using `sqlCriteria` but it might be because I did not want the overhead of fingerprint computation which is a derived field of `sqlCriteria` anyway. We had enough hotspot on fingerprint computation that introducing one here did not make sense.